src/PlaneWaveBasis.jl

```diff
@@ -113,7 +119,8 @@ Base.Broadcast.broadcastable(basis::PlaneWaveBasis) = Ref(basis)
 Base.eltype(::PlaneWaveBasis{T}) where {T} = T
 
 @timing function build_kpoints(model::Model{T}, fft_size, kcoords, Ecut;
-                               variational=true) where {T}
+                               variational=true,
+                               architecture::AbstractArchitecture=CPU()) where {T}
```

Comment:
I think I would make the `architecture` a non-optional kwarg (i.e. remove the `=CPU()`).